sound/soc/sof/control.c

```diff
@@ -229,14 +229,16 @@ int snd_sof_bytes_get(struct snd_kcontrol *kcontrol,
 		return -EINVAL;
 	}
 
-	size = data->size + sizeof(*data);
-	if (size > be->max) {
+	/* be->max has been verified to be >= sizeof(struct sof_abi_hdr) */
+	if (data->size > be->max - sizeof(*data)) {
 		dev_err_ratelimited(scomp->dev,
-				    "error: DSP sent %zu bytes max is %d\n",
-				    size, be->max);
+				    "error: DSP claims to send %u bytes of data max is %lu\n",
+				    data->size, be->max - sizeof(*data));
```

Comment:
@lyakh this message is not quite right. There is actually no DSP involved here at all. We're only passing the cached value in the control